src/main/java/fi/hsl/transitdata/pubtransredisconnect/Main.java

```diff
@@ -272,21 +272,30 @@ private void handleJourneyResultSet(ResultSet resultSet, Jedis jedis) throws Exc
             values.put(TransitdataProperties.KEY_OPERATING_DAY, resultSet.getString(OPERATING_DAY));
 
             String key = TransitdataProperties.REDIS_PREFIX_DVJ + resultSet.getString(DVJ_ID);
-            jedis.hmset(key, values);
-            jedis.expire(key, redisTTLInSeconds);
-
-            //Insert a composite key that allows reverse lookup of the dvj id
-            //The format is route-direction-date-time
-            String joreKey = TransitdataProperties.formatJoreId(resultSet.getString(ROUTE_NAME),
-                    resultSet.getString(DIRECTION), resultSet.getString(OPERATING_DAY),
-                    resultSet.getString(START_TIME));
-            jedis.set(joreKey, resultSet.getString(DVJ_ID));
-            jedis.expire(joreKey, redisTTLInSeconds);
-
-            rowCounter++;
+            String response = jedis.hmset(key, values);
+            if (checkRedisResponse(response)) {
+                jedis.expire(key, redisTTLInSeconds);
+                rowCounter++;
+
+                //Insert a composite key that allows reverse lookup of the dvj id
+                //The format is route-direction-date-time
+                String joreKey = TransitdataProperties.formatJoreId(resultSet.getString(ROUTE_NAME),
+                        resultSet.getString(DIRECTION), resultSet.getString(OPERATING_DAY),
+                        resultSet.getString(START_TIME));
+                response = jedis.set(joreKey, resultSet.getString(DVJ_ID));
+                if (checkRedisResponse(response)) {
+                    jedis.expire(joreKey, redisTTLInSeconds);
+                    rowCounter++;
+                } else {
+                    log.error("Failed to set joreKey {}, Redis returned {}", joreKey, response);
+                }
+            } else {
+                log.error("Failed to set Trip details {}, Redis returned {}", key, response);
+            }
+
         }
 
-        log.info("Inserted " + rowCounter + " dvj keys");
+        log.info("Inserted " + rowCounter + " dvj and reverse-lookup keys");
```

Comment:
Do we combine the counting?

Reply:
yea now the rowCounter is incremented after each redis insert. how do you feel, which approach is better?

Reply:
Well, at least one can see there is something wrong if the number is not even :) How about logging an error if number of DB rows != dvjId (!= joreId)?

Reply:
Yea good idea for separating the counters, implemented. Perhaps the extra error is not needed since redis-failure is the only way for the counters not to match and that is already flagged.